cmd/kubeadm/app/cmd/phases/preflight.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"github.com/spf13/cobra"
+
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	"k8s.io/kubernetes/cmd/kubeadm/app/preflight"
+)
+
+func NewCmdPreFlight() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "preflight",
+		Short: "Run pre-flight checks",
+		RunE:  subCmdRunE("preflight"),
+	}
+
+	cmd.AddCommand(NewCmdPreFlightMaster())
+	cmd.AddCommand(NewCmdPreFlightNode())
+	return cmd
+}
+
+func NewCmdPreFlightMaster() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "master",
+		Short: "Run master pre-flight checks",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg := &kubeadmapi.MasterConfiguration{}
```

Comment:
Ideally, I think `preflight` should indeed accept all the same parameters that `init` accepts, to simulate the same environment that `init` will have when running. I can open a new issue for that.